### DISCUSSION TITLE
Speaking multiple languages in a single recording only transcribes the english parts?

### DISCUSSION THREAD

**Discussion by Jamie452** [deepgram/community#761] [General help] [answered]
Using Nova-2, if I speak in multiple languages in a single audio file, it seems to only pick out and transcribe the English, ignoring the other languages.

1) Is there a way to improve this already?

2) Can this be addressed in a future model?

```
await this.deepgram.listen.prerecorded.transcribeFile(file.buffer, {
     model: "nova-2",
     detect_language: true
     smart_format: true,
     diarize: true,
})
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment, marked as answer]
@Jamie452  👋  Great question currently Nova-2 can't process multi-lingual audio files but this support of language or "code" switching is something we are actively working on.  I don't have an ETA of when that feature would be available though.

If you're dealing with audio that contains multiple languages, you'd need to segment the audio based on the language spoken and then transcribe each segment separately with the appropriate language model.
